lib/transaction.js

```diff
@@ -280,13 +280,7 @@ function makeTransactor(trx, connection, trxClient) {
       return trxClient.transaction(container, options, trx);
     } else {
       return new Promise((resolve, _reject) => {
-        trxClient.transaction(
-          (nestedTrx) => {
-            resolve(nestedTrx);
-          },
-          options,
-          trx
-        );
+        trxClient.transaction(resolve, options, trx).catch(_reject);
```

Comment:
in which case was this triggered? is it difficult to simulate in a test?

Reply:
Hmm... I can probably come up w/ a test case for it.  I'll give it a try in a few minutes.

Reply:
Thanks for reviewing all of these, btw!  Any hesitations before we merge this change?